DoLah/Parser/DateTimeParser.cpp

```diff
@@ -56,44 +58,93 @@ namespace DoLah {
         std::string out;
         for (size_t m = 0; m < monthPattern.size(); m++) {
             if (std::regex_match(str, std::regex(monthPattern.at(m), std::regex_constants::icase))) {
-                return (int) m;
+                return (int)m;
             }
         }
 
-        return NULL;
+        return REJECT;
     }
 
     int DateTimeParser::getYear(std::string str) {
-        if (str.length() != 2 && str.length() != 4) {
-            return NULL;
-        } else if (!ParserLibrary::isDecimal(str)) {
-            return NULL;
+        std::string year = str;
+        if (year.length() != 2 && year.length() != 4) {
+            return REJECT;
+        } else if (!ParserLibrary::isDecimal(year)) {
+            return REJECT;
         }
 
-        return std::stoi(str);
+        if (year.length() == 2) {
+            year = CENTURY + year;
+        }
+
+        return std::stoi(year);
     }
 
+
+
     std::tm DateTimeParser::toDateFormat(std::vector<std::string> strArr) {
+        std::tm output;
+
+        std::vector<std::string> cleanArr = strArr;
+        if (cleanArr.size() == 1) {
+            std::string str = cleanArr.at(0);
+            for (size_t i = 0; i < dateDividers.size(); i++) {
+                cleanArr = ParserLibrary::explode(str, dateDividers.at(i));
+                if (cleanArr.size() > 1) {
+                    break;
+                }
+            }
+        }
+        cleanArr = ParserLibrary::removeElementsFromStringVector(cleanArr, decorators);
+
+        try {
+            output = classifyDate(cleanArr);
+        } catch (std::invalid_argument e) {
+            throw e;
+        }
+
+        return output;
+    }
+
+    std::string DateTimeParser::tmToString(std::tm time) {
+        return std::to_string(time.tm_mday) + "/" + std::to_string(time.tm_mon + 1) + "/" + std::to_string(time.tm_year + 1900);
```

Comment:
I removed the method bc it was solely for unit test. Unit test itself has this part now. It's like this:

    dates += std::to_string(datesVector.at(i).tm_mday) + "/" 
                        + std::to_string(datesVector.at(i).tm_mon + 1) + "/" 
                        + std::to_string(datesVector.at(i).tm_year + 1900);